boxoffice/views/order.py

```diff
@@ -0,0 +1,198 @@
+
+    user = User.query.filter_by(email=buyer_form.email.data).first()
+    order = Order(user=user,
+        organization=organization,
+        item_collection=item_collection,
+        buyer_email=buyer_form.email.data,
+        buyer_fullname=buyer_form.fullname.data,
+        buyer_phone=buyer_form.phone.data)
+
+    line_item_tups = LineItem.calculate([{'item_id': li_form.data.get('item_id')}
+        for li_form in line_item_forms
+        for _ in range(li_form.data.get('quantity'))], coupons=discount_coupons)
+
+    for idx, line_item_tup in enumerate(line_item_tups):
+        item = Item.query.get(line_item_tup.item_id)
+        if line_item_tup.discount_policy_id:
+            policy = DiscountPolicy.query.get(line_item_tup.discount_policy_id)
+        else:
+            policy = None
+        if line_item_tup.discount_coupon_id:
+            coupon = DiscountCoupon.query.get(line_item_tup.discount_coupon_id)
+        else:
+            coupon = None
+
+        line_item = LineItem(order=order, item=item, discount_policy=policy,
+            line_item_no=idx+1,
+            discount_coupon=coupon,
+            ordered_at=datetime.utcnow(),
+            base_amount=line_item_tup.base_amount,
+            discounted_amount=line_item_tup.discounted_amount,
+            final_amount=line_item_tup.base_amount-line_item_tup.discounted_amount)
+        if coupon:
+            coupon.register_use()
+            db.session.add(coupon)
+        db.session.add(line_item)
+
+    db.session.add(order)
+    db.session.commit()
+    return jsonify(code=200, order_id=order.id,
+        order_access_token=order.access_token,
+        payment_url=url_for('payment', order=order.id),
+        free_order_url=url_for('free', order=order.id),
+        final_amount=order.get_amounts().final_amount)
+
+
+@app.route('/<order>/free', methods=['GET', 'OPTIONS', 'POST'])
+@load_models(
+    (Order, {'id': 'order'}, 'order')
+    )
+@xhr_only
+@cross_origin(origins=ALLOWED_ORIGINS)
+def free(order):
+    """
+    Completes a order which has a final_amount of 0
+    """
+    order_amounts = order.get_amounts()
+    if order_amounts.final_amount == 0:
+        order.confirm_sale()
+        db.session.add(order)
+        db.session.commit()
+        return jsonify(code=200)
+    else:
+        return api_result(402, 'Free order confirmation failed')
+
+
+@app.route('/<order>/payment', methods=['GET', 'OPTIONS', 'POST'])
+@load_models(
+    (Order, {'id': 'order'}, 'order')
+    )
+@xhr_only
+@cross_origin(origins=ALLOWED_ORIGINS)
+def payment(order):
+    """
+    Accepts JSON containing `pg_payment_id`.
+
+    Creates a payment object, attempts to 'capture' the payment from Razorpay,
+    and returns a JSON containing the result of the operation.
+
+    A successful capture results in a `payment_transaction` registered against the order.
+    """
+    pg_payment_id = request.json.get('pg_payment_id')
+    if not (request.json and pg_payment_id):
```

Comment:
You should be checking `request.json` before attempting to read from it (previous line).